scripts: ci: check_compliance strips required information

**Describe the bug**

The error parsing regex in `check_compliance.py` strips out information required to fix the problem for `duplicate-code` warnings:  https://github.com/zephyrproject-rtos/zephyr/blob/8a2a44f9e98c313038f59ad3ceef857af58c5892/scripts/ci/check_compliance.py#L950

**Logs and console output**

For code triggering the similarity checker, `check_compliance.py` outputs the following:
```
Running Nits             tests in /Users/user/code/workspace/repo ...
Running Pylint           tests in /Users/user/code/workspace/repo ...
/Users/user/code/workspace/zephyr/scripts/repontrc
pylint --rcfile=/Users/user/code/workspace/zephyr/scripts/ci/pylintrc --load-plugins=argparse-checker scripts/build_release.py scripts/util/add_release_metadata.py
::warning file=scripts/util/add_release_metadata.py,line=1,col=0,title=R0801::scripts/util/add_release_metadata.py:1 Similar lines in 2 files
Running ModulesMaintainers tests in /Users/user/code/workspace/repo ...
Skipping KconfigBasic
Running Checkpatch       tests in /Users/user/code/workspace/repo ...
Running ImageSize        tests in /Users/user/code/workspace/repo ...
1 checks failed
ERROR   : Test Pylint failed:
R0801:Similar lines in 2 files
File:scripts/util/add_release_metadata.py
Line:1
Column:0

Complete results in compliance.xml
```
and `compliance.xml` contains:
```
<?xml version="1.0" encoding="utf-8"?>
<testsuites tests="12" failures="1" errors="0" skipped="0" time="0.0">
	<testsuite name="Compliance" tests="12" errors="0" failures="1" skipped="0" time="0">
		<testcase name="GitDiffCheck" classname="Guidelines"/>
		<testcase name="YAMLLint" classname="Guidelines"/>
		<testcase name="DevicetreeBindings" classname="Guidelines"/>
		<testcase name="Identity" classname="Guidelines"/>
		<testcase name="MaintainersFormat" classname="Guidelines"/>
		<testcase name="Gitlint" classname="Guidelines"/>
		<testcase name="BinaryFiles" classname="Guidelines"/>
		<testcase name="Nits" classname="Guidelines"/>
		<testcase name="Pylint" classname="Guidelines">
			<failure message="scripts/util/add_release_metadata.py:1 Similar lines in 2 files" type="warning">
R0801:Similar lines in 2 files
File:scripts/util/add_release_metadata.py
Line:1
Column:0</failure>
		</testcase>
		<testcase name="ModulesMaintainers" classname="Guidelines"/>
		<testcase name="Checkpatch" classname="Guidelines"/>
		<testcase name="ImageSize" classname="Guidelines"/>
	</testsuite>
</testsuites>
```

Neither of which are helpful for determining which files have similar code, or where in the file that code is.
It actually points to the wrong location (line 1, instead of line 48).
Running the pylint command manually on the same failing code outputs:
```
************* Module add_release_metadata
scripts/util/add_release_metadata.py:1:0: R0801: Similar lines in 2 files
==add_release_metadata:[48:66]
==build_release:[107:126]
    boot = edt.label2node['boot_partition']
    app = edt.label2node['slot0_partition']
    scratch = edt.label2node.get('firmware_patch_partition', None)
    partitions = {
        'bootloader': {
            'offset': boot.regs[0].addr,
            'size': boot.regs[0].size,
        },
        'application': {
            'offset': app.regs[0].addr,
            'size': app.regs[0].size,
        },
    }
    if scratch is not None:
        partitions['scratch'] = {'size': scratch.regs[0].size}
    return partitions
```

--- a/scripts/ci/check_compliance.py
+++ b/scripts/ci/check_compliance.py
@@ -7,6 +7,7 @@
 import argparse
 import collections
 from email.utils import parseaddr
+import json
 import logging
 import os
 from pathlib import Path
@@ -177,7 +178,6 @@ class ComplianceTest:
         fail = FmtdFailure(severity, title, file, line, col, desc)
         self._result(fail, fail.text)
         self.fmtd_failures.append(fail)
-
 
 class EndTest(Exception):
     """
@@ -1169,7 +1169,7 @@ class PyLint(ComplianceTest):
         else:
             python_environment["PYTHONPATH"] = check_script_dir
 
-        pylintcmd = ["pylint", "--rcfile=" + pylintrc,
+        pylintcmd = ["pylint", "--output-format=json2", "--rcfile=" + pylintrc,
                      "--load-plugins=argparse-checker"] + py_files
         logger.info(cmd2str(pylintcmd))
         try:
@@ -1181,21 +1181,19 @@ class PyLint(ComplianceTest):
                            env=python_environment)
         except subprocess.CalledProcessError as ex:
             output = ex.output.decode("utf-8")
-            regex = r'^\s*(\S+):(\d+):(\d+):\s*([A-Z]\d{4}):\s*(.*)$'
-
-            matches = re.findall(regex, output, re.MULTILINE)
-            for m in matches:
-                # https://pylint.pycqa.org/en/latest/user_guide/messages/messages_overview.html#
+            messages = json.loads(output)['messages']
+            for m in messages:
                 severity = 'unknown'
-                if m[3][0] in ('F', 'E'):
+                if m['messageId'][0] in ('F', 'E'):
                     severity = 'error'
-                elif m[3][0] in ('W','C', 'R', 'I'):
+                elif m['messageId'][0] in ('W','C', 'R', 'I'):
                     severity = 'warning'
-                self.fmtd_failure(severity, m[3], m[0], m[1], col=m[2],
-                        desc=m[4])
+                self.fmtd_failure(severity, m['messageId'], m['path'],
+                                  m['line'], col=str(m['column']), desc=m['message']
+                                  + f" ({m['symbol']})")
 
-            # If the regex has not matched add the whole output as a failure
-            if len(matches) == 0:
+            if len(messages) == 0:
+                # If there are no specific messages add the whole output as a failure
                 self.failure(output)
 
 

